cli/src/commands/new.rs

```diff
@@ -134,68 +137,69 @@ Please use `jj new 'all:x|y'` instead of `jj new --allow-large-revsets x y`.",
-            )?;
-        }
+        children_commits = target_commits;
+    } else if args.insert_after {
+        parent_commits = target_commits;
+        parent_commit_ids = parent_commits.iter().ids().cloned().collect();
+        let parents_expression = RevsetExpression::commits(target_ids);
+        // Each child of the targets will be rebased: its set of parents will be updated
+        // so that the targets are replaced by the new commit.
+        // Exclude children that are ancestors of the new commit
+        let children_expression = parents_expression
+            .children()
+            .minus(&parents_expression.ancestors());
+        children_commits = children_expression
+            .evaluate_programmatic(workspace_command.repo().as_ref())?
+            .iter()
+            .commits(workspace_command.repo().store())
+            .try_collect()?;
+        workspace_command.check_rewritable(children_commits.iter().ids())?;
     } else {
-        let old_parents = RevsetExpression::commits(target_ids.clone());
-        let commits_to_rebase: Vec<Commit> = if args.insert_after {
-            // Each child of the targets will be rebased: its set of parents will be updated
-            // so that the targets are replaced by the new commit.
-            // Exclude children that are ancestors of the new commit
-            let to_rebase = old_parents.children().minus(&old_parents.ancestors());
-            to_rebase
-                .evaluate_programmatic(tx.base_repo().as_ref())?
-                .iter()
-                .commits(tx.base_repo().store())
-                .try_collect()?
-        } else {
-            vec![]
-        };
-        tx.base_workspace_helper()
-            .check_rewritable(commits_to_rebase.iter().ids())?;
-        let merged_tree = merge_commit_trees(tx.repo(), &target_commits)?;
-        new_commit = tx
-            .mut_repo()
-            .new_commit(command.settings(), target_ids.clone(), merged_tree.id())
-            .set_description(join_message_paragraphs(&args.message_paragraphs))
-            .write()?;
-        num_rebased = commits_to_rebase.len();
-        for child_commit in commits_to_rebase {
-            let commit_parents = RevsetExpression::commits(child_commit.parent_ids().to_owned());
-            let new_parents = commit_parents.minus(&old_parents);
-            let mut new_parent_ids = new_parents
-                .evaluate_programmatic(tx.base_repo().as_ref())?
-                .iter()
-                .collect_vec();
-            new_parent_ids.push(new_commit.id().clone());
-            rebase_commit(
-                command.settings(),
-                tx.mut_repo(),
-                child_commit,
-                new_parent_ids,
-            )?;
-        }
+        parent_commits = target_commits;
+        parent_commit_ids = parent_commits.iter().ids().cloned().collect();
+        children_commits = vec![];
+        workspace_command.check_rewritable([])?;
```

Comment:
nit: perhaps, `check_rewritable()` can be factored out. (can be addressed later)

Reply:
I added a new TODO there. I think it isn't strictly necessary for `jj new x` to check for immutability, but this breaks an existing test case which has this assumption.

Reply:
Oops, I meant `check_rewritable(children_commits)` could be moved out of `if { .. } else { .. }`.

Anyway, it can be changed later. Some tests might fail if we reorder error handling.

Reply:
Thanks, I misunderstood your comment. I shifted the check out of the if-else.

I was referring to this existing test case: https://github.com/martinvonz/jj/blob/fcc0b86f0a7ff2990dc9e7e42a4e3e12eacdbb86/cli/tests/test_immutable_commits.rs#L60-L68

I'm not sure if all commands which create new commits have at minimum a  `workspace_command.check_rewritable([])` check which checks if the `immutable_heads()` revset is correctly defined.

Reply:
> https://github.com/martinvonz/jj/blob/fcc0b86f0a7ff2990dc9e7e42a4e3e12eacdbb86/cli/tests/test_immutable_commits.rs#L60-L68
> 
> I'm not sure if all commands which create new commits have at minimum a `workspace_command.check_rewritable([])` check which checks if the `immutable_heads()` revset is correctly defined.

Interesting. I don't think it's mandatory requirement, but might be nice to have in `jj new`.

Btw, the test added at 88a4a8281fd5. As far as I can tell, the command to produce an error wouldn't be important, and it's probably okay to use `jj edit` instead.